packages/@sfx/core/src/utils/dependencies.ts

```diff
@@ -0,0 +1,97 @@
+import { Plugin } from '../plugin';
+
+/**
+ * The type of a plugin dependency graph. This dependency graph
+ * represents the reverse dependency relationships of the plugins: a
+ * plugin points to the plugins that list it as a dependency. For
+ * example, plugin A will point to plugin B if B lists A as a
+ * dependency.
+ *
+ * The graph is represented as an adjacency list, which itself is
+ * represented as an object whose keys are plugin names and whose values
+ * are lists of the names of the depending plugins.
+ */
+export interface DependencyGraph {
+  /** The names of each plugin's dependers. */
+  [name: string]: string[];
+}
+
+/**
+ * Creates an empty dependency graph.
+ *
+ * @returns An empty dependency graph.
+ * @hidden
+ */
+function createEmptyGraph(): DependencyGraph {
+  return Object.create(null);
+}
+
+/**
+ * Builds a dependency graph from the given plugins.
+ *
+ * @param plugins The plugins from which to build the graph.
+ * @returns The completed dependency graph.
+ */
+export function createDependencyGraph(plugins: Plugin[] = []): DependencyGraph {
+  const dependencyGraph = createEmptyGraph();
+
+  plugins.forEach(({ metadata: { name, depends } }) => {
+    if (!dependencyGraph[name]) dependencyGraph[name] = [];
```

Comment:
If we use the dependency graph for any analysis later on, it's a lot simpler down the line for the caller to be able to assume that a plugin exists rather than having to check that the plugin exists in the graph before checking its dependers. It's also a simpler invariant to say that the graph contains all plugins rather than all plugins that have dependers.